Cache slow/disabled test files

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55682 Cache slow/disabled test files**

Fixes #55648

For now it downloads and writes the relevant files to the system's temp dir and marks it as valid for 3 hours.

Differential Revision: [D27685616](https://our.internmc.facebook.com/intern/diff/D27685616)

Commit:
Cache slow/disabled test files

Fixes #55648

For now it downloads and writes the relevant files to the system's temp
dir and marks it as valid for 1 hour.

[ghstack-poisoned]

Commit:
Update on "Cache slow/disabled test files"


Fixes #55648

For now it downloads and writes the relevant files to the system's temp
dir and marks it as valid for 1 hour.

[ghstack-poisoned]

janeyx99 commented on torch/testing/_internal/common_utils.py:
```diff
@@ -817,17 +819,44 @@ def settings(*args, **kwargs):
+        now = datetime.datetime.now()
+        mtime = datetime.datetime.fromtimestamp(fname.stat().st_mtime)
+        diff = now - mtime
+        return diff.total_seconds() < datetime.timedelta(hours=1).seconds
```
Maybe do 3 hours as most tests run longer than 1 hour? And it may be a good idea to store the number of hours in a global variable like `FILE_CACHE_LIFESPAN` for more visibility and clarity.

janeyx99 commented on torch/testing/_internal/common_utils.py:
```diff
@@ -817,17 +819,44 @@ def settings(*args, **kwargs):
-        except Exception as e:
-            print(f'Could not download slow test stats because of error {e}. Proceeding with no added slow tests.')
-            slow_tests_dict = {}
+        url = "https://raw.githubusercontent.com/pytorch/test-infra/master/stats/.pytorch-slow-tests"
```
This url recently changed to https://raw.githubusercontent.com/pytorch/test-infra/master/stats/slow-tests.json, just fyi

janeyx99 reviewed (approved):
Change looks good overall. I think a good next step would be to prevent the code from processing the disabled tests over and over again as well.

Commit:
Update on "Cache slow/disabled test files"


Fixes #55648

For now it downloads and writes the relevant files to the system's temp dir and marks it as valid for 1 hour.

[ghstack-poisoned]

Commit:
Update on "Cache slow/disabled test files"


Fixes #55648

For now it downloads and writes the relevant files to the system's temp dir and marks it as valid for 1 hour.

[ghstack-poisoned]

Commit:
Update on "Cache slow/disabled test files"


Fixes #55648

For now it downloads and writes the relevant files to the system's temp dir and marks it as valid for 1 hour.

[ghstack-poisoned]

Commit:
Update on "Cache slow/disabled test files"


Fixes #55648

For now it downloads and writes the relevant files to the system's temp dir and marks it as valid for 3 hours.

[ghstack-poisoned]
